File: Codes from the AI-assistants/[Experimental] mandelbrot/Basic Prompt - Normal Mode/Mandelbrot-deepseek-basic-normal.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

typedef struct {
    double real;
    double imag;
} Complex;

int mandelbrot_iterations(Complex c, int max_iter) {
    Complex z = {0.0, 0.0};
    int iter = 0;
    
    while (iter < max_iter) {
        double z_real_sq = z.real * z.real;
        double z_imag_sq = z.imag * z.imag;
        
        if (z_real_sq + z_imag_sq > 4.0) {
            break;
        }
        
        double new_real = z_real_sq - z_imag_sq + c.real;
        double new_imag = 2.0 * z.real * z.imag + c.imag;
        
        z.real = new_real;
        z.imag = new_imag;
        iter++;
    }
    
    return iter;
}

void generate_mandelbrot_pbm(const char* filename, int width, int height, 
                            double x_min, double x_max, double y_min, double y_max, 
                            int max_iter) {
    FILE* file = fopen(filename, "wb");
    if (!file) {
        perror("Failed to open file");
        return;
    }
    
    // Write PBM header
    fprintf(file, "P1\n%d %d\n", width, height);
    
    // Generate Mandelbrot set
    for (int y = 0; y < height; y++) {
        for (int x = 0; x < width; x++) {
            Complex c;
            c.real = x_min + (x_max - x_min) * x / (width - 1);
            c.imag = y_min + (y_max - y_min) * y / (height - 1);
            
            int iter = mandelbrot_iterations(c, max_iter);
            
            // Output 1 for points in Mandelbrot set, 0 for points outside
            fprintf(file, "%d ", (iter == max_iter) ? 1 : 0);
        }
        fprintf(file, "\n");
    }
    
    fclose(file);
}

int main() {
    int width = 800;
    int height = 600;
    int max_iter = 1000;
    
    // Define the region to plot
    double x_min = -2.0;
    double x_max = 1.0;
    double y_min = -1.0;
    double y_max = 1.0;
    
    generate_mandelbrot_pbm("mandelbrot.pbm", width, height, 
                           x_min, x_max, y_min, y_max, max_iter);
    
    printf("Mandelbrot set saved to mandelbrot.pbm\n");
    return 0;
}

File: Codes from the AI-assistants/[Experimental] mandelbrot/Basic Prompt - Normal Mode/Mandelbrot-deepseek-basic-normal.py
import array
import math

def mandelbrot_iterations(c, max_iter):
    """
    Calculate the number of iterations for a complex point c
    Returns the number of iterations until escape or max_iter if in set
    """
    z = complex(0, 0)
    for i in range(max_iter):
        if abs(z) > 2.0:
            return i
        z = z * z + c
    return max_iter

def generate_mandelbrot_pbm(filename, width, height, x_min, x_max, y_min, y_max, max_iter):
    """
    Generate a PBM file of the Mandelbrot set
    """
    with open(filename, 'wb') as f:
        # Write PBM header
        header = f"P1\n{width} {height}\n"
        f.write(header.encode('ascii'))
        
        # Generate Mandelbrot data
        for y in range(height):
            row = array.array('B')
            im = y_min + (y_max - y_min) * y / (height - 1)
            
            for x in range(width):
                re = x_min + (x_max - x_min) * x / (width - 1)
                c = complex(re, im)
                
                iter_count = mandelbrot_iterations(c, max_iter)
                
                # 1 for points in Mandelbrot set, 0 for points outside
                pixel = 1 if iter_count == max_iter else 0
                row.append(pixel)
            
            # Write row as space-separated ASCII values
            row_str = ' '.join(map(str, row)) + '\n'
            f.write(row_str.encode('ascii'))

def main():
    width = 800
    height = 600
    max_iter = 1000
    
    # Define the region to plot
    x_min, x_max = -2.0, 1.0
    y_min, y_max = -1.0, 1.0
    
    generate_mandelbrot_pbm("mandelbrot.pbm", width, height, 
                           x_min, x_max, y_min, y_max, max_iter)
    
    print("Mandelbrot set saved to mandelbrot.pbm")

if __name__ == "__main__":
    main()